packages/vscode-extension/src/connectionService.ts

```diff
@@ -44,52 +51,79 @@ export function getConnection(key: string): Connection | null {
 export async function deleteConnection(key: string): Promise<void> {
   const connections = getConnections();
   const connection = connections[key];
-  if (connection) {
-    delete connections[key];
-    await updateConnections(connections);
-    await deletePassword(key);
-    await sendNotification('connectionDeleted', {
-      ...connection,
-      connect: false,
-    });
+  if (!connection) {
+    return;
   }
+
+  delete connections[key];
+  await updateConnections(connections);
+  await deletePassword(key);
+  await updateDbmsConnection('connectionDeleted', {
+    ...connection,
+    connect: false,
+  });
 }
 
-export async function saveConnection(
+export function saveConnection(
+  connection: Connection | null,
+): Promise<ConnnectionResult>;
+export function saveConnection(
   connection: Connection | null,
   password: string,
-): Promise<void> {
+): Promise<ConnnectionResult>;
+
+export async function saveConnection(
+  connection: Connection | null,
+  password?: string,
+): Promise<ConnnectionResult> {
   if (!connection) {
     return;
   }
 
-  const connections = disconnectAllAndGetConnections();
+  const connections = connection.connect
+    ? disconnectAllAndGetConnections()
+    : getConnections();
+
   connections[connection.key] = connection;
 
   await updateConnections(connections);
```

Comment:
Sorry I know this is not from this pr, but why do we need this line? I thought if connections were a reference and we've already updated it, we wouldn't need to update the global state?